src/main/scala/com/lightbend/kafkalagexporter/PrometheusEndpointSink.scala

```diff
@@ -12,34 +12,41 @@ import io.prometheus.client.{CollectorRegistry, Gauge}
 import scala.util.Try
 
 object PrometheusEndpointSink {
-  def apply(httpPort: Int, definitions: MetricDefinitions): MetricsSink =
-    Try(new PrometheusEndpointSink(httpPort, definitions))
+  def apply(appConfig: AppConfig, definitions: MetricDefinitions): MetricsSink =
+    Try(new PrometheusEndpointSink(appConfig, definitions))
       .fold(t => throw new Exception("Could not create Prometheus Endpoint", t), sink => sink)
 }
 
-class PrometheusEndpointSink private(httpPort: Int, definitions: MetricDefinitions) extends MetricsSink {
+class PrometheusEndpointSink private(appConfig: AppConfig, definitions: MetricDefinitions) extends MetricsSink {
 
-  private val server = new HTTPServer(httpPort)
+  private val server = new HTTPServer(appConfig.port)
   private val registry = CollectorRegistry.defaultRegistry
 
   DefaultExports.initialize()
 
-  private val metrics: Map[GaugeDefinition, Gauge] = definitions.map(definition =>
-    definition -> Gauge.build()
-      .name(definition.name)
-      .help(definition.help)
-      .labelNames(definition.labels: _*)
-      .register(registry)
-  ).toMap
+  private val metrics: Map[String, Map[GaugeDefinition, Gauge]] = {
+    appConfig.clusters.map { cluster =>
+      val globalLabelNamesForCluster = appConfig.globalLabelsForCluster(cluster.name).map(_._1).toSeq
```

Comment:
Could you match the tuple instead of using the shortcut methods (i.e. `_._1`).  If that's a `Map` then you could get the keys with the `keys`.